tools/json2cbor/json2cbor.c

```diff
@@ -328,13 +328,14 @@ CborError decode_json(cJSON *json, CborEncoder *encoder)
         err = cbor_encode_double(encoder, json->valuedouble);
 
         if (err == CborErrorOutOfMemory) {
+            ptrdiff_t offset = cbor_encoder_get_buffer_size(&container, buffer);
             buffersize += 1024;
             uint8_t *newbuffer = realloc(buffer, buffersize);
             if (newbuffer == NULL)
```

Comment:
This is definitely true for the current state of ``json2cbor`` code but sometimes code evolves and error handling paths change. I'm also pretty sure that ``decode_json`` and other functions here are useful on their own (and will be reused by someone either separately or as a part of some code that is based on the whole source of this tool, I saw similar situations), so IMHO it makes sense to analyze functions separately too and purge inconsistencies that caught some attention before they spread. Just my 2¢.

Thank you for your time and kind reply anyway.

Reply:
Fair enough. It's a simple change to make and this is not performance critical, so I'll make it.

Reply:
O-oh, I'm sorry for confusion but it looks like ``encoder->end`` became invalid after commit [2e2efde](https://github.com/intel/tinycbor/pull/262/commits/2e2efdecb679166c7c380ac33ae639611b331859) (it's the latest version of your commit, isn't it?). That's why I suggested to introduce the ``newbuffersize`` variable to avoid copy-paste bugs (DRY principle rises an alarm inside me when I see duplicate ``+ 1024`` in different parts of the same code) and usage of stale values. Another solution is to move ``buffersize += 1024;`` before ``encoder->end = newbuffer + buffersize;``

Did I misunderstood anything?

Reply:
Right, this is wrong now.

Reply:
Looks good now, thanks!